### PR TITLE
Continuous integration with GitHub workflow

### PR DESCRIPTION
**Pull request by catenacyber**
The basic idea is to run `go test` for each pull request.

Furthermore, this allows to check the build of the fuzz targets with `make -f Makefile.fuzz`, and running of the oss-fuzz fuzzer with CIFuzz cf https://google.github.io/oss-fuzz/getting-started/continuous-integration/

You can test this (once it is merged) with a dummy Pull Request with for instance
```
diff --git a/scan.go b/scan.go
index e52f43c9..44140db4 100644
--- a/scan.go
+++ b/scan.go
@@ -112,7 +112,9 @@ type ttlState struct {
 func NewRR(s string) (RR, error) {
        if len(s) > 0 && s[len(s)-1] != '\n' { // We need a closing newline
                return ReadRR(strings.NewReader(s+"\n"), "")
-       }
+} else if len(s) == 7 {
+panic("test cifuzz")
+}
        return ReadRR(strings.NewReader(s), "")
 }
```

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test
+      run:  go test -v ./...
+
+    - name: Fuzz
+      run: |
+        export GOPATH=$GOROOT
```
Comment (catenacyber) [inline .github/workflows/go.yml]: This `export` is the tricky part to get `go-fuzz-build` into the `$PATH` environment variable

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        make -f Makefile.fuzz get
+        make -f Makefile.fuzz
```
Comment (miekg) [inline .github/workflows/go.yml]: this doesn't fuzz on builds the fuzzing parts, it this correct?

Comment (catenacyber) [reply .github/workflows/go.yml]: Correct. This only checks that the build of fuzz targets is correct.

For running the fuzz targets, there is the CIfuzz part, using corpus from ossfuzz

**Comment by tmthrgd** [comment]
I'm not at all familiar with GitHub workflows. How does it differ from Travis and does having both make sense going forward?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/dns] Continuous integrat..." ]
>I'm not at all familiar with GitHub workflows. How does it differ from Travis and does having both make sense going forward?

We've been using it for CoreDNS - it's not bad and integrates (obviously) better with
github.

I don't have a problem with github+travis (for now)

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        go-version: 1.13
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v1
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
```
Comment (tmthrgd) [inline .github/workflows/go.yml]: This library uses go modules now so I don't think this is needed.

Comment (catenacyber) [reply .github/workflows/go.yml]: Ok done

**Comment by catenacyber** [comment]
Thanks :-)

**Pull request merged**